src/lib.rs

```diff
@@ -229,6 +302,68 @@ impl IRCompiler {
         }
     }
 
+    #[allow(clippy::too_many_arguments)]
+    pub fn set_ray_tracing_pipeline_arguments(
+        &mut self,
+        max_attribute_size_in_bytes: u32,
+        raytracing_pipeline_flags: ffi::IRRaytracingPipelineFlags,
+        chs: u64,
+        miss: u64,
+        any_hit: u64,
+        callable_args: u64,
+        max_recursive_depth: i32,
+    ) {
+        unsafe {
+            self.funcs.IRCompilerSetRayTracingPipelineArguments(
+                self.me,
+                max_attribute_size_in_bytes,
+                raytracing_pipeline_flags,
+                chs,
+                miss,
+                any_hit,
+                callable_args,
+                max_recursive_depth,
+            )
+        }
+    }
+
+    pub fn set_hitgroup_type(&mut self, hit_group_type: ffi::IRHitGroupType) {
+        unsafe {
+            self.funcs
+                .IRCompilerSetHitgroupType(self.me, hit_group_type)
+        }
+    }
+
+    pub fn synthesize_indirect_intersection_function(
+        &mut self,
+    ) -> Result<IRMetalLibBinary, CompilerError> {
+        let target_metallib = IRMetalLibBinary::new(self);
+        if unsafe {
+            self.funcs
+                .IRMetalLibSynthesizeIndirectIntersectionFunction(self.me, target_metallib.me)
+        } {
+            return Ok(target_metallib);
```

Comment:
Drop `return`, same below.

Reply:
```
error[E0308]: mismatched types
   --> src/lib.rs:355:13
    |
349 | /         if unsafe {
350 | |             self.funcs.IRMetalLibSynthesizeIndirectIntersectionFunction(
351 | |                 self.me.as_ptr(),
352 | |                 target_metallib.me.as_ptr(),
...   |
355 | |             Ok(target_metallib)
    | |             ^^^^^^^^^^^^^^^^^^^ expected `()`, found `Result<IRMetalLibBinary, _>`
356 | |         }
    | |_________- expected this to be `()`
    |
    = note: expected unit type `()`
                    found enum `Result<IRMetalLibBinary, _>`
help: you might have meant to return this value
    |
355 |             return Ok(target_metallib);
    |             ++++++                    +
    ```

Reply:
Ah this is where I forgot the else statement